### PR TITLE
[CLEANUP] Upgrading karma, jasmine and phantomjs to match other pentaho projects

### PR DESCRIPTION
**Pull request by davidmsantos90**
Also enables us to test javascript ES5 code

@nantunes can you give a call to the build team to see if they are ok with this?
@graimundo @dcleao @carlosrusso @marcovala please review

### CODE REVIEW

File: config/karma.ci.conf.js [hunk 1]
```diff
@@ -37,8 +37,18 @@ module.exports = function(config) {
...
-        dir : 'bin/coverage/reports/'
+      reporters: [
+        {
+          type: "html",
+          dir:  "bin/reports/jscoverage/html/"
+        },
+        {
+          type: "cobertura",
+          dir:  "bin/reports/cobertura/xml/"
+        }
```
Comment (carlosrusso) [inline config/karma.ci.conf.js]: Not that it matters much, but do we really use the xml coverage reports? I certainly never have.

Comment (nantunes) [inline config/karma.ci.conf.js]: Aren't they used by Jenkins at devci.pentaho.com? And by sonar? And by wingman?

Comment (carlosrusso) [inline config/karma.ci.conf.js]: That would make sense. Thanks!

**Comment by nantunes** [comment]
I would go ahead and merge. The build team hasn't raised any meaningful concern.

**Comment by nantunes** [comment]
@graimundo @dcleao @carlosrusso @marcovala @davidmsantos90 ⬆

**Pull request merged**